src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -894,7 +894,13 @@ protected void ParseSshHostName(string hostname, out string host, out string use
             try
             {
                 Uri uri = new System.Uri("ssh://" + hostname);
-                host = ResolveComputerName(uri.Host);
+
+                // Extract original hostname from URI with preserved case
+                // This is needed since System.Uri canonicalizes URI and makes host lowercase
+                int originalHostnameIndex = uri.OriginalString.IndexOf(uri.Host, StringComparison.OrdinalIgnoreCase);
```

Comment:
Don't believe this will ever return `-1` for `IndexOf()` but maybe we need a safeguard and set to default if it does?

Reply:
Why not use `hostname`?

Reply:
`hostname` can include port and username, in this case this is just extracting the host part.